### ISSUE TITLE
Bug about export C++ to Es6

### ISSUE THREAD

**Issue by Linkersem** [emscripten-core/emscripten#21160]
I compiled my c++ code into an ES6 module through Emscripten, but after I imported it via "import", when I ran npm run build, it kept showing out of memory, I tried to increase the memory, but no matter how big I increased, it always reported an error exceeding memory, my compilation settings are as follows, what should I do to solve this problem?

This is the setting of the cmakelist.txt for the entire project：

```
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O3 -Wall -fPIC -pthread ")
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1 -fexceptions")
	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s MODULARIZE=1 -s EXPORT_ES6=1 -s USE_ES6_IMPORT_META=1")
	set(CMAKE_CXX_STANDARD 11)
	add_definitions(-DWASM_JS)
```

This is the setting for the submodule's cmakelist.txt：

```
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  -O3 -s MODULARIZE=1 -s SINGLE_FILE=0 -s EXPORT_NAME='MTMP' -s DEMANGLE_SUPPORT=1")
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  -pthread -s USE_PTHREADS=1 -s PTHREAD_POOL_SIZE=4 -s ASYNCIFY -s ASSERTIONS -s PTHREAD_POOL_SIZE_STRICT=0 ")
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  -sSTACK_SIZE=10485760 --memory-init-file 0 -s TOTAL_MEMORY=256MB -s WASM_MEM_MAX=1GB -s ALLOW_MEMORY_GROWTH=1 ")
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  -s EXPORT_EXCEPTION_HANDLING_HELPERS -fexceptions ")
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  --bind -s EXPORTED_RUNTIME_METHODS=['ccall','getValue','setValue','cwrap'] ")
set(EMSCRIPTEN_LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}  -o ${js_name}.js")
set_target_properties(${js_name} PROPERTIES LINK_FLAGS "${EMSCRIPTEN_LINK_FLAGS}")
```

The following error message is displayed：

![image](https://github.com/emscripten-core/emscripten/assets/49711774/586adb68-ecab-48e0-b5b7-9076b8470882)

**Comment by kripken** [comment]
`JavaScript heap out of memory` is an error from the JS VM itself. No changes in Emscripten can help with that problem. The JS VM is just running out of memory for JS objects.

It's possible your machine does not have enough RAM to run the program?

**Comment by sbc100** [comment]
Don't node itself have some internal maximum that can be increases?  Or will it just keep allocating until the OS itself runs out of memory?

**Comment by Linkersem** [comment]
> `JavaScript heap out of memory` is an error from the JS VM itself. No changes in Emscripten can help with that problem. The JS VM is just running out of memory for JS objects.
> 
> It's possible your machine does not have enough RAM to run the program?

Yes, my memory is sufficient, I have 32G of memory, and I've tried with this command ”NODE_OPTIONS=--max-old-space-size“ to set the available memory to 10G, but I still get the problem of out of memory

**Comment by Linkersem** [comment]
> 节点本身难道没有一些可以增加的内部最大值吗？还是会一直分配，直到操作系统本身内存不足？

From what I understand, it has a default maximum, but I also tried to increase the memory to ~10g, and he still has the problem of out of memory; But the strange thing is that I don't specify that when my code is compiled to ES6, this library will work in my test example, but the modularity used in my new project is ES6, I don't know how I should import it correctly if I don't compile my code to ES6?

**Comment by kripken** [comment]
If changing ES6 mode determines whether this breaks or not, that is probably worth looking into. Perhaps you can use the node.js heap/memory profiler to see what is different in the two cases.
